### ISSUE TITLE
CORS error when fetch image from S3

### ISSUE THREAD

**Issue by AkiraNoob** [cyntler/react-doc-viewer#71]
![image](https://user-images.githubusercontent.com/89231623/196373593-0c63c32d-85f8-4117-bb42-421d7744e067.png)
![image](https://user-images.githubusercontent.com/89231623/196373768-4b06f350-edec-4dc3-9a26-f2efabe0fb02.png)
![image](https://user-images.githubusercontent.com/89231623/196374135-aae1e739-922c-49db-9f09-31d326fd929e.png)

*note: ImageDisplay is my custom image display

I faced this issue while working in my own project although in S3 i set Accpet-Control-Allow-Origin to '*' (image 3). I found out that if i write a fetch method to that s3 url its still face the same error but when i set mode: 'no-cors' it success. Anyone can explain why for me :( is there any way to work around or to set no-cors mode to reac-doc-viewer?

This is my first time submit an issue so plz appologize for any mistakes, thank you guys a lot.

**Comment by jamesmart77** [comment]
TLDR; You are doing a cross-origin request, even though `no-cors` is set. That config simply allows the request to go through, but you'll still see the error (which is actually a warning).

---------------

I've experienced a similar issue on a different project. By performing a fetch from your client, from your DocViewer, you are doing a cross-origin call. This is typically against common practice because of the potential security vulnerabilities that are invited. 

However, if you have the correct controls in place, this can be perfectly legitimate. If your scenario was anything like mine, the `fetch` was perfectly fine, but the errors in the console are more so an FYI, "Be advised...", type of thing.

Check out the [MDN Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#:~:text=mode%3A%20%27cors%27%2C%20//%20no%2Dcors%2C%20*cors%2C%20same%2Dorigin) resource.

**Comment by DenkyOgbonnaya** [comment]
Adding prefetchMethod="GET" resolved the CORS issue for me.

**Comment by belferink1996** [comment]
I'm experiencing this too.
I have a valid URL from AWS S3, which fetches fine when pasted in the browser.

But when Doc Viewer tries to make the request, it fails on CORS and no file is shown (infinite loading).
I tried the above of adding `prefetchMethod="GET"`, it did not resolve the issue.

**Comment by AkiraNoob** [comment]
hi guys, sorry for late reply, i have solved it by remove region prefix in s3 presigned URL, maybe u can try 
![image](https://github.com/cyntler/react-doc-viewer/assets/89231623/1f71f686-57b7-4774-9bb0-30ea1effa53b)

**Comment by Jova266** [comment]
> I'm experiencing this too. I have a valid URL from AWS S3, which fetches fine when pasted in the browser.
> 
> But when Doc Viewer tries to make the request, it fails on CORS and no file is shown (infinite loading). I tried the above of adding `prefetchMethod="GET"`, it did not resolve the issue.

Have you managed to resolve this issue? I have the same one

**Comment by ermarkar** [comment]
anybody got the solution? same url works if img tag i use

**Comment by BenElferink** [comment]
> > I'm experiencing this too. I have a valid URL from AWS S3, which fetches fine when pasted in the browser.
> > But when Doc Viewer tries to make the request, it fails on CORS and no file is shown (infinite loading). I tried the above of adding `prefetchMethod="GET"`, it did not resolve the issue.
> 
> Have you managed to resolve this issue? I have the same one @BenElferink

Nope, still unresolved.

**Comment by ermarkar** [comment]
> > > I'm experiencing this too. I have a valid URL from AWS S3, which fetches fine when pasted in the browser.
> > > But when Doc Viewer tries to make the request, it fails on CORS and no file is shown (infinite loading). I tried the above of adding `prefetchMethod="GET"`, it did not resolve the issue.
> > 
> > 
> > Have you managed to resolve this issue? I have the same one @BenElferink
> 
> Nope, still unresolved.

i exposed a new api from my server and it worked, but still signed urls not working

`
@ApiBearerAuth()
  @Get("/docs/preview")
  async getEnquiryDoc(@Query('key') key: string, @Res() res: any): Promise<any> {
    try {
      const s3Object = await this.s3Servcice.getObject(key);
      // Determine the content type based on the file extension or your specific logic
      const contentType = 'application/octet-stream';

      res.setHeader('Content-Disposition', `attachment; filename=${key}`);
      res.setHeader('Content-Type', contentType);
      res.setHeader('Content-Length', s3Object.length);
      res.send(s3Object);

    } catch (err) {
      throw err;
    }
  }

/**
   * Get an S3 object based on the provided key.
   * @param {string} key The object key (filename) in the S3 bucket.
   * @returns {Promise<Buffer>} The content of the S3 object as a Buffer.
   */
  async getObject(key: string): Promise<Buffer> {
    const params = ({
      Bucket: this.bucketName,
      Key: key,
    });

    const response = await this.s3.getObject(params).promise();
    return response.Body as Buffer;
  }
`
